### DISCUSSION TITLE
如何运行内置宏获取msvc版本？

### DISCUSSION THREAD

**Discussion by WilliamMiaoYH** [xmake-io/xmake#3782] [Q&A]
我现在想实现对某一个编译器的最低要求版本的判断。系统里有vs2019和vs2022。
要求最低支持vs2019，所以在全局切换到vs2019之后，在rules获取
`local vs = import("core.tool.toolchain").load("msvc"):config("vs")`
输出的还是2022

我了解到，msvc提供了_MSC_VER内置宏，请问怎么在xmake里去获取该值呢？
clang、gcc，倒是可以使用`find_tool("clang", {version = true}）`去获取判断，
我执行`find_tool("link", {version = true})`没有任何效果
类似cmake里的去限制MSVC最低版本
```
elseif(CMAKE_CXX_COMPILER_ID MATCHES MSVC)
  if(MSVC_VERSION VERSION_LESS "1928")
```
谢谢🙏

**Comment by waruqi** [comment]
target:toolchain("msvc"):config("vs")

**Comment by jingkaimori** [comment]
> `target:toolchain("msvc"):config("vs")`

如果获取mingw工具链的版本，能不能用`config`这种方法？

用`detect.sdks.find_mingw`拿到的工具链对象只有路径没有版本信息

**Reply by waruqi** [reply]
目前没有 mingw 版本信息
